worlds/tits_the_3rd/client/client.py

```diff
@@ -100,13 +229,59 @@ async def tits_the_3rd_watcher(ctx: TitsThe3rdContext):
             continue
 
         if ctx.game_interface.should_write_world_player_identifier():
+            logger.info(f"Game Start Dectected. Setting up AP Verification Hook")
             ctx.game_interface.write_world_player_identifier(ctx.world_player_identifier)
+            logger.info(f"AP Verification Hook Set. You can now save and resume the game for this AP Seed")
             continue
 
+        try:
+            if ctx.exit_event.is_set():
+                break
+            if not ctx.server:
+                continue
+            if ctx.game_interface.should_send_and_recieve_items(ctx.world_player_identifier):
+                await check_for_locations(ctx)
+
+            if ctx.game_interface.is_valid_to_receive_item() and ctx.game_interface.should_send_and_recieve_items(wpid=ctx.world_player_identifier):
+                await ctx.send_item()
+
+            if ctx.game_interface.is_valid_to_receive_item() and ctx.game_interface.should_send_and_recieve_items(wpid=ctx.world_player_identifier):
+                await ctx.give_item()
+
+        except Exception as err:
+            logger.warning("*******************************")
+            logger.warning("Encountered error. Please post a message to the thread on the AP discord")
+            logger.warning("*******************************")
+            logger.exception(str(err))
+            # attempt to reconnect at the top of the loop
+            continue
+
+
+async def check_for_locations(ctx: TitsThe3rdContext):
```

Comment:
Lets stick this as a method under the game context instead. Idk why I did it like this for rabi ribi.